homeassistant/components/slack/__init__.py

```diff
@@ -1,2 +1,68 @@
-"""The slack component."""
-DOMAIN = "slack"
+"""The slack integration."""
+import logging
+
+from aiohttp.client_exceptions import ClientError
+from slack import WebClient
+from slack.errors import SlackApiError
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_PLATFORM, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, discovery
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DATA_CLIENT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.NOTIFY]
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Slack component."""
+    # Iterate all entries for notify to only get Slack
+    if Platform.NOTIFY in config:
+        for entry in config[Platform.NOTIFY]:
+            if entry[CONF_PLATFORM] == DOMAIN:
+                hass.async_create_task(
+                    hass.config_entries.flow.async_init(
+                        DOMAIN, context={"source": SOURCE_IMPORT}, data=entry
+                    )
+                )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Slack from a config entry."""
+    session = aiohttp_client.async_get_clientsession(hass)
+    slack = WebClient(token=entry.data[CONF_API_KEY], run_async=True, session=session)
+
+    try:
+        await slack.auth_test()
+    except SlackApiError as ex:
+        if ex.response["error"] == "invalid_auth":
+            _LOGGER.error("Invalid API key")
+            return False
+        raise ConfigEntryNotReady("Error while setting up integration") from ex
+    except ClientError as ex:
+        _LOGGER.warning(
```

Comment:
Why continue here and not retry later?

Reply:
This mirrors the behavior before this PR

Reply:
Does that still make sense?

Reply:
@bachya What do you think?

Reply:
Not sure why that was there originally but agreed w/ @bdraco: if we can't connect to the Slack API, we should raise `ConfigEntryNotReady`.